theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
Yes it's not what confuses me, when I said any access to x, I meant any access to x where x is a CudaNdarray. I'm confused because normally to have any operation going on the GPU we have some c code with __global__ functions meant exclusively for the GPU (be it hidden under PyCuda or anything else).